rust/abacus-core/src/traits/indexer.rs

```diff
@@ -7,18 +7,24 @@
 //! traits (CommonIndexer or OutboxIndexer) to provide an common interface which
 //! other entities can retrieve this chain-specific info.
 
+use std::fmt::Debug;
+
 use async_trait::async_trait;
 use eyre::Result;
 
-use crate::{CheckpointWithMeta, RawCommittedMessage};
+use crate::{CheckpointWithMeta, InterchainGasPaymentWithMeta, RawCommittedMessage};
 
-/// Interface for Abacus Common contract indexer. Interface that allows for other
-/// entities to retrieve chain-specific data from an outbox or inbox.
+/// Interface for an indexer.
 #[async_trait]
-pub trait AbacusCommonIndexer: Send + Sync + std::fmt::Debug {
+pub trait Indexer: Send + Sync + Debug {
     /// Get chain's latest block number
```

Comment:
It's counterintuitive to me that we wouldn't distinguish between "latest block number" (in the comment) and "finalized_block_number" (for the function name). Finalized makes me expect to see some lag measured in number of blocks